### PR TITLE
feat!: add LSP17Extendable to LSP7 and LSP8

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?

## 🚀 Feature

- Add LSP17Extendable to LSP7 and LSP8

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol [hunk 1]
```diff
@@ -43,14 +57,118 @@ abstract contract LSP7DigitalAsset is
...
+    ) internal virtual override returns (bytes memory) {
+        // If there is a function selector
+        address extension = _getExtension(msg.sig);
+
+        // if no extension was found, revert
+        if (extension == address(0))
+            revert NoExtensionFoundForFunctionSelector(msg.sig);
+
+        (bool success, bytes memory result) = extension.call{value: msg.value}(
+            abi.encodePacked(callData, msg.sender, msg.value)
```
Comment (skimaharvey) [inline contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol:118]: do you think it still makes sense to append the `msg.value` when we can directly access it?

File: contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol [hunk 2]
```diff
@@ -43,14 +57,118 @@ abstract contract LSP7DigitalAsset is
...
+    ) internal view virtual override returns (address) {
+        // Generate the data key relevant for the functionSelector being called
+        bytes32 mappedExtensionDataKey = LSP2Utils.generateMappingKey(
+            _LSP17_EXTENSION_PREFIX,
+            functionSelector
+        );
+
+        // Check if there is an extension stored under the generated data key
+        address extension = address(
+            bytes20(ERC725YCore._getData(mappedExtensionDataKey))
```
Comment (skimaharvey) [inline contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol:151]: i guess sanity check is not necessary

File: contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol (hunk 1 above)
Comment (YamenMerhi) [reply contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol:118]: For consistency across extensions yes

**Review by skimaharvey — APPROVED** [review]
Lgtm

**Pull request merged**